cli/src/pcluster/models/imagebuilder.py

```diff
@@ -146,6 +157,91 @@ def config(self):
             self.__config = ImageBuilderSchema().load(self.source_config)
         return self.__config
 
+    @property
+    def bucket(self):
+        """Return a bucket configuration."""
+        if self.__bucket is None:
+            self.__bucket = S3BucketFactory.init_s3_bucket(
+                service_name=self.image_name,
+                stack_name=self.image_name,
+                custom_s3_bucket=self._get_custom_bucket(),
+                artifact_directory=self.s3_artifacts_dir,
+            )
+        return self.__bucket
+
+    def _get_custom_bucket(self):
```

Comment:
optional: we can transform this into a get_bucket_name and return whatever is in the tag and then have the caller check if the returned name is == to the S3Bucket.get_bucket_name so that we simplify all this articulated logic

Reply:
what if stack doesn't exist? custom_bucket_name only needs during initalization, we can use name property to get bucket name in other cases, so I think it's more reasonable to put custom bucket name retrieving in imagebuilder/cluster.

Reply:
done